### DISCUSSION TITLE
Good practice: One global store, or separate stores for separate concerns?

### DISCUSSION THREAD

**Discussion by bjmc** [pmndrs/zustand#2486] [Q&A]
Hi, we're using Zustand in a React application and we need to store some "business logic data" (for lack of a better term, this is data the app will fetch from an API, manipulate, and resubmit) but then there is also some "UI preference data" (light/dark theme, local user settings, etc.)

My first impulse was that these are two completely orthogonal types of information, so they belong in two separate Zustand stores, but then the Zustand documentation says:

> [Recommended patterns](https://docs.pmnd.rs/zustand/guides/flux-inspired-practice#recommended-patterns)
> [Single store](https://docs.pmnd.rs/zustand/guides/flux-inspired-practice#single-store)
>
> Your applications global state should be located in a single Zustand store.
>
> If you have a large application, Zustand supports [splitting the store into slices](https://docs.pmnd.rs/zustand/guides/slices-pattern).

Our app is not particularly large, and we'd never (that I can see) need to update both stores as part of one user action, so it doesn't seem like the slice pattern applies here.

What would folks recommend for a situation like this?

**Comment by dbritto-dev** [comment]
IMHO is something like this:

- Big store with independent data between each other: multiple stores
- Big store with dependent data between each other: multiple slices
- Medium and small store: one store

**Reply by bjmc** [reply]
Right now, the user prefs are pretty small, so maybe it's simpler to put it all in one store. I suppose we could always refactor to two stores later, if it becomes too large or confusing.
